test/middleware-mockserver.js

```diff
@@ -0,0 +1,277 @@
+/* eslint-env node */
+var url = require( "url" );
+var fs = require( "fs" );
+var getRawBody = require( "raw-body" );
+
+var cspLog = "";
+var mocks = {
```

Comment:
This file and mock.php should cross-reference each other to minimize divergence.